src/Hooks.ts

```diff
@@ -16,9 +16,21 @@ export function useNavigation<S>(): NavigationScreenProp<S & NavigationRoute> {
 }
 
 export function useNavigationParam<T extends keyof NavigationParams>(
-  paramName: T
+  paramName: T,
+  fallbackValue?: NavigationParams[T]
 ) {
-  return useNavigation().getParam(paramName);
+  const { getParam, setParams } = useNavigation();
+  const setParamValue = useCallback(
+    () => {
+      return (newValue: NavigationParams[T]) =>
+        setParams({ [paramName]: newValue });
+    },
+    [setParams]
+  );
+  return [
+    fallbackValue ? getParam(paramName, fallbackValue) : getParam(paramName),
+    setParamValue(),
+  ];
```

Comment:
Both `setParams` in deps of `useCallback` (https://github.com/react-navigation/core/issues/71) and redundant function wrapping causes crash with "Maximum update depth exceeded" error in my React-Native app, so suggest to avoid it. Also added `paramName` prop as `ESLint` suggested
```
const setParamValue = useCallback(
    (newValue: NavigationParams[T]) => setParams({ [paramName]: newValue }),
    // `setParams` function is not memoized, so that it causes "Maximum update depth exceeded" (https://github.com/react-navigation/core/issues/71)
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [paramName],
  );
  return [
    fallbackValue ? getParam(paramName, fallbackValue) : getParam(paramName),
    setParamValue,
  ];
```